File: 01_extracao/verificar_extracao.py
"""
VERIFICAÇÃO PÓS-EXTRAÇÃO
Roda após download_bigquery.py para validar os dados antes do tratamento.
"""

from pathlib import Path
import pandas as pd

ARQUIVO = Path("data/processed/rais_economistas.parquet")

if not ARQUIVO.exists():
    print("Arquivo não encontrado. Rode download_bigquery.py primeiro.")
    exit()

df = pd.read_parquet(ARQUIVO)

DESCRICAO_CBO = {
    "251205": "Economista",
    "251210": "Economista agrícola",
    "251215": "Economista industrial",
    "251220": "Economista do trabalho",
    "251225": "Economista do setor público",
    "251230": "Economista ambiental",
    "251235": "Economista regional e urbano",
}

print("=" * 55)
print("RELATÓRIO DE VERIFICAÇÃO — RAIS ECONOMISTAS")
print("=" * 55)

print(f"\nDimensões: {len(df):,} vínculos × {df.shape[1]} colunas")

print("\nCobertura temporal:")
print(df["ano"].value_counts().sort_index().to_string())

print("\nCódigos CBO presentes:")
cbo_counts = df["cbo_2002"].value_counts().reset_index()
cbo_counts.columns = ["cbo_2002", "n"]
cbo_counts["descricao"] = cbo_counts["cbo_2002"].map(DESCRICAO_CBO).fillna("Outro")
cbo_counts["pct"] = (cbo_counts["n"] / len(df) * 100).round(1).astype(str) + "%"
print(cbo_counts.to_string(index=False))

print("\nDistribuição por sexo:")
print(df["sexo"].value_counts().to_string())

print("\nTop 10 UFs:")
print(df["sigla_uf"].value_counts().head(10).to_string())

print("\nMissings por variável:")
missings = df.isnull().sum()
missings = missings[missings > 0]
if missings.empty:
    print("  Nenhum missing encontrado.")
else:
    print(missings.to_string())

print("\nEstatísticas de remuneração (vl_remun_media_nom > 0):")
remun = df.loc[df["valor_remuneracao_media"] > 0, "valor_remuneracao_media"]
print(remun.describe().round(2).to_string())

print("\n" + "=" * 55)
print("Verificação concluída. Se os dados parecerem")
print("coerentes, prossiga para clean_rais.py")
print("=" * 55)

File: 01_extracao/verificar_extracao.R
# =============================================================================
# VERIFICAÇÃO PÓS-EXTRAÇÃO
# Script: verificar_extracao.R
# Objetivo: validar integridade dos dados baixados antes de prosseguir
#           para o estágio de tratamento (clean_rais.R)
# =============================================================================

library(arrow)
library(dplyr)
library(tidyr)
library(ggplot2)

# --- 1. Leitura --------------------------------------------------------------

DIR_PROCESSED <- here::here("data", "processed")

df <- arrow::read_parquet(
  file.path(DIR_PROCESSED, "rais_economistas.parquet")
)

# --- 2. Verificações básicas -------------------------------------------------

cat("========================================\n")
cat("RELATÓRIO DE VERIFICAÇÃO - RAIS\n")
cat("========================================\n\n")

cat("Dimensões:\n")
cat("  Linhas  :", scales::comma(nrow(df)), "\n")
cat("  Colunas :", ncol(df), "\n\n")

cat("Cobertura temporal:\n")
print(table(df$ano))

cat("\nCódigos CBO presentes na extração:\n")
df |>
  count(cbo_2002, sort = TRUE) |>
  mutate(
    descricao = case_when(
      cbo_2002 == "251205" ~ "Economista",
      cbo_2002 == "251210" ~ "Economista agrícola",
      cbo_2002 == "251215" ~ "Economista industrial",
      cbo_2002 == "251220" ~ "Economista do trabalho",
      cbo_2002 == "251225" ~ "Economista do setor público",
      TRUE                 ~ "Outro/não mapeado"
    ),
    pct = scales::percent(n / sum(n), accuracy = 0.1)
  ) |>
  print()

cat("\nDistribuição por sexo:\n")
df |>
  count(sexo) |>
  mutate(pct = scales::percent(n / sum(n), accuracy = 0.1)) |>
  print()

cat("\nDistribuição por UF (top 10):\n")
df |>
  count(sigla_uf, sort = TRUE) |>
  slice_head(n = 10) |>
  print()

cat("\nMissings por variável:\n")
df |>
  summarise(across(everything(), ~ sum(is.na(.)))) |>
  tidyr::pivot_longer(everything(),
                      names_to  = "variavel",
                      values_to = "n_missing") |>
  filter(n_missing > 0) |>
  mutate(pct_missing = scales::percent(n_missing / nrow(df), accuracy = 0.01)) |>
  print()

cat("\nEstatísticas de remuneração (vínculos com remun > 0):\n")
df |>
  filter(vl_remun_media_nom > 0) |>
  summarise(
    min    = min(vl_remun_media_nom, na.rm = TRUE),
    p25    = quantile(vl_remun_media_nom, 0.25, na.rm = TRUE),
    mediana = median(vl_remun_media_nom, na.rm = TRUE),
    media  = mean(vl_remun_media_nom, na.rm = TRUE),
    p75    = quantile(vl_remun_media_nom, 0.75, na.rm = TRUE),
    max    = max(vl_remun_media_nom, na.rm = TRUE)
  ) |>
  mutate(across(everything(), scales::comma)) |>
  print()

cat("\n========================================\n")
cat("Verificação concluída. Se os totais e\n")
cat("distribuições parecerem razoáveis,\n")
cat("prossiga para clean_rais.R\n")
cat("========================================\n")
